### PR TITLE
Update forward-backward passes

### PR DESCRIPTION
**Pull request by frapac**
- remove dead code
- catch properly exception when no solution is being found by the solver
- update tests for SDDP
- fix benchmark.jl

### CODE REVIEW

**Comment by leclere** [comment]
Looks good to me.

File: examples/benchmark.jl [hunk 1]
```diff
@@ -197,19 +197,17 @@ function benchmark_sddp()
     # Launch benchmark
```
Comment (Henri-Gerard) [inline examples/benchmark.jl:197]: I think we should change one of file benchmark.jl I d'ont know which one

**Comment by Henri-Gerard** [comment]
Ok for me

File: examples/benchmark.jl (hunk 1 above)
Comment (frapac) [inline examples/benchmark.jl:197]: Maybe we could do that once you merge your PR

**Pull request merged**